src/resources/assets/js/geo/components/filterByLocationComponent.vue

```diff
@@ -33,17 +26,22 @@ export default {
     data() {
         return {
             selectedItem: null,
-            trigger: false,
+            showModal: false,
         };
     },
     methods: {
-        submit() {
-            // always trigger rule-refresh in geoFIlter.js (in case the rule has already been applied)
-            // not regarding whether sessionStorage-data has been changed or not 
-            window.dispatchEvent(customEvent);
-            // selectedItem is always null, so geoFilter-rule can only be added once
+        submit(key) {
+            this.hideModal();
+            // pass the array of selected IDs as JSON-string in order to work properly in the hasRule()-check later on.
+            // hasRule() function checks whether rule-attributes diverge between existing and new rules
+            // e.g. newRule.data === oldRule.data (does not assert equality for array- or object-comparison)
+            let ids = JSON.parse(sessionStorage.getItem(key)) || [];
+            this.selectedItem = JSON.stringify(ids);
```

Comment:
Why the detour through session storage? Is there a way to return the data directly from the modal (e.g. instead of `key` here)?

If the data persists in session storage, the images will be selected again when the modal opens up the next time. I think this is counter-intuitive since we allow multiple selections now.